### PR TITLE
chore: remove beta experimental flag

### PR DESCRIPTION
**Pull request by mingkun2020**
#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/logs/command.py [hunk 1]
```diff
@@ -163,9 +150,6 @@ def do_cli(
     from samcli.lib.utils.boto_utils import get_boto_client_provider_with_config, get_boto_resource_provider_with_config
 
     if not names or len(names) > 1:
```
Comment (mndeveci) [inline samcli/commands/logs/command.py:152]: I think this if statement needs some adjustments right? It was either asking for beta feature opt in or printing an informational message.

File: samcli/commands/sync/command.py [hunk 2]
```diff
@@ -77,20 +70,8 @@
...
-This feature is currently in beta. Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.
-
-The SAM CLI will use the AWS Lambda, Amazon API Gateway, and AWS StepFunctions APIs to upload your code without 
-performing a CloudFormation deployment. This will cause drift in your CloudFormation stack. 
-**The sync command should only be used against a development stack**.
-
-Confirm that you are synchronizing a development stack and want to turn on beta features.
-
-Enter Y to proceed with the command, or enter N to cancel:
-"""
```
Comment (mndeveci) [inline samcli/commands/sync/command.py:90]: We still want this confirmation in the beginning of `sam sync`. So we only need to remove first line here, rest should stay there and we should get a confirmation from customer every time they run `sam sync`.

**Review by mndeveci — CHANGES_REQUESTED** [review]

File: samcli/commands/sync/command.py (hunk 2 above)
Comment (mingkun2020) [reply samcli/commands/sync/command.py:90]: We already have one name `SYNC_CONFIRMATION_TEXT` above, just remove this beta one.

Comment (mndeveci) [reply samcli/commands/sync/command.py:90]: Got it, I thought we had one text and we display it all the time. Thanks!

File: samcli/commands/sync/command.py [hunk 3]
```diff
@@ -244,15 +224,9 @@ def do_cli(
 
     confirmation_text = SYNC_CONFIRMATION_TEXT
```
Comment (mndeveci) [inline samcli/commands/sync/command.py:225]: nit: then we can remove this line here by directly using `SYNC_CONFIRMATION_TEXT` below

File: samcli/commands/logs/command.py (hunk 1 above)
Comment (mingkun2020) [reply samcli/commands/logs/command.py:152]: This if statement check if the name is not provided we will print the hint. In the previous beta version we check if the beta flag is off then we don't print the hint then exist but now we don't need the beta check anymore.

Comment (mndeveci) [reply samcli/commands/logs/command.py:152]: But there was an `else` statement there which was checking;
- If no names or multiple names provided, ask for beta opt-in
- Else print informational message that `sam logs` can be used without any name flag.

So now it seems like it is printing the information message if no names or multiple names are provided. It should be opposite.

Comment (mingkun2020) [reply samcli/commands/logs/command.py]: I see. Thanks for the explanation. Updated the logic.

**Pull request merged**